pilot/pkg/security/authz/builder/builder.go

```diff
@@ -15,113 +15,141 @@
 )
 
-// Builder wraps all needed information for building the RBAC filter for a service.
+// Builder builds Istio authorization policy to Envoy RBAC filter.
 type Builder struct {
-	generator policy.Generator
+	trustDomainBundle trustdomain.Bundle
+	denyPolicies      []model.AuthorizationPolicyConfig
+	allowPolicies     []model.AuthorizationPolicyConfig
 }
 
-// NewBuilder creates a builder instance that can be used to build corresponding RBAC filter config.
-func NewBuilder(trustDomainBundle trustdomain.Bundle, workloadLabels labels.Collection, configNamespace string,
+// New returns a new builder for the given workload with the authorization policy.
+// Returns nil if none of the authorization policies are enabled for the workload.
+func New(trustDomainBundle trustdomain.Bundle, workload labels.Collection, namespace string,
 	policies *model.AuthorizationPolicies) *Builder {
-	var generator policy.Generator
-
-	denyPolicies, allowPolicies := policies.ListAuthorizationPolicies(configNamespace, workloadLabels)
-	if len(denyPolicies) > 0 || len(allowPolicies) > 0 {
-		generator = v1beta1.NewGenerator(trustDomainBundle, denyPolicies, allowPolicies)
-		rbacLog.Debugf("found authorization policies for workload %v in %s", workloadLabels, configNamespace)
-	}
-
-	if generator == nil {
+	denyPolicies, allowPolicies := policies.ListAuthorizationPolicies(namespace, workload)
+	if len(denyPolicies) == 0 && len(allowPolicies) == 0 {
 		return nil
 	}
-
 	return &Builder{
-		generator: generator,
+		trustDomainBundle: trustDomainBundle,
+		denyPolicies:      denyPolicies,
+		allowPolicies:     allowPolicies,
 	}
 }
 
-// BuildHTTPFilters build RBAC HTTP filters.
-func (b *Builder) BuildHTTPFilters() []*httpFilterPb.HttpFilter {
-	if b == nil {
-		return nil
-	}
+// BuilderHTTP returns the RBAC HTTP filters built from the authorization policy.
+func (b Builder) BuildHTTP() []*httppb.HttpFilter {
+	var filters []*httppb.HttpFilter
 
-	var filters []*httpFilterPb.HttpFilter
-	denyConfig, allowConfig := b.generator.Generate(false /* forTCPFilter */)
-	if denyFilter := createHTTPFilter(denyConfig); denyFilter != nil {
-		filters = append(filters, denyFilter)
+	if denyConfig := build(b.denyPolicies, b.trustDomainBundle,
+		false /* forTCP */, true /* forDeny */); denyConfig != nil {
+		filters = append(filters, createHTTPFilter(denyConfig))
 	}
-	if allowFilter := createHTTPFilter(allowConfig); allowFilter != nil {
-		filters = append(filters, allowFilter)
+	if allowConfig := build(b.allowPolicies, b.trustDomainBundle,
+		false /* forTCP */, false /* forDeny */); allowConfig != nil {
+		filters = append(filters, createHTTPFilter(allowConfig))
 	}
+
 	return filters
 }
 
-// nolint: interfacer
-func createHTTPFilter(config *envoyRbacHttpPb.RBAC) *httpFilterPb.HttpFilter {
-	if config == nil {
-		return nil
-	}
+// BuildTCP returns the RBAC TCP filters built from the authorization policy.
+func (b Builder) BuildTCP() []*tcppb.Filter {
```

Comment:
[nit] I actually like the old name better as it's more descriptive.

Reply:
You mean `BuildTCPFilters()`? Hmm, we can change that later.